firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -83,26 +83,37 @@ void home()
 // Moves the magnet from the "start" point to the "end" point
 // This can only move in straight lines
 // Returns a boolean indicating success/error
-bool moveStraight(int motor[], int startCol, int startRow, int endCol, int endRow)
+bool moveStraight(int motor[], float startCol, float startRow, float endCol, float endRow)
 {
   // A specific motor is passed to this function since we are only moving one here
 
   // How many steps per space
-  int spaces, dir, numSteps;
+  float pieceSpaces;
+  int dir, numSteps;
   int i;
 
+  // Makes sure all arugments produce a multiple of unitspaces
+  if (fmodf(startCol, 0.5) != 0)
+    return false;
+  else if (fmodf(startRow, 0.5) != 0)
+    return false;
+  else if (fmodf(endCol, 0.5) != 0)
+    return false;
+  else if (fmodf(endRow, 0.5) != 0)
+    return false;
+
```

Comment:
So this can be improved. Since I showed you this code this is my bad, but here's why. Here we always check 4 if-statements. Remember, if statements are considered expensive. We can remove this if we just OR all of the conditions together. What I'm looking for is a structure similar to what is on line 118 of uart.ino.